JuanManuelGarcia.cpp

```diff
@@ -0,0 +1,412 @@
+		cout << "El paciente es: " << paciente->getNombre() << endl;
+	}
+	string getFecha()
+	{
+		return fecha;
+	}
+	int getCosto()
+	{
+		return costo;
+	}
+
+
+};
+class AtencionClinica : public Atencion
+{
+private:
+
+public:
+	AtencionClinica(string tipo1, string fecha1, int hora1, Paciente* paciente1, int costo1) : Atencion( tipo1,  fecha1,  hora1,  paciente1,  costo1)
+	{
+		this->tipo = "Clinica";
+	}
+	void mostrarAtencionClinica()
+	{
+		cout << "La fecha es: " << fecha << endl;
+		cout << "La hora es: " << hora << endl;
+		cout << "El paciente es: " << paciente << endl;
+		cout << "El costo es: " << costo << endl;
+
+	}
+};
+class AtencionDomicilio : public Atencion
+{
+private:
+
+public:
+	AtencionDomicilio(string tipo1, string fecha1, int hora1, Paciente* paciente1, int costo1) : Atencion(tipo1, fecha1, hora1, paciente1, costo1)
+	{
+		this->tipo = "Domicilio";
+	}
+	void mostrarAtencionClinica()
+	{
+		cout << "La fecha es: " << fecha << endl;
+		cout << "La hora es: " << hora << endl;
+		cout << "El paciente es: " << paciente << endl;
+		cout << "El costo es: " << costo << endl;
+
+	}
+};
+class VectorAtencion
+{
+private:
+	int cantMax;
+	int cantActual;
+	Atencion** atenciones;
+	int costoTotal;
+public:
+	VectorAtencion(int cantMax)
+	{
+		this->cantMax = cantMax;
+		this->cantActual = 0;
+		this->atenciones = new Atencion * [cantMax];
+	}
+	void ordenarPorFecha()
+	{
+		Atencion* aux;
+		for (int i = 0; i < cantActual; i++)
+		{
+			for (int j = 0; j < cantActual-1; j++)
+			{
+				if (atenciones[j]->getFecha() > atenciones[j + 1]->getFecha())
+				{
+					aux = atenciones[j];
+					atenciones[j] = atenciones[j + 1];
+					atenciones[j + 1] = aux;
+				}
+			}
+		}
+	}
+	void mostrarAtenciones()
+	{
+		cout << "=== LISTA DE Atenciones ===" << endl;
+		for (int i = 0; i < cantActual; i++)
+		{
+			cout << i + 1 << ". ";
+			atenciones[i]->mostrarAtencion();
+		}
+	}
+	void calcularCostoAtenciones()
+	{
+		for(int i = 0; i < cantActual; i++)
+		{
+
+			atenciones[i]->calcularCosto();
+		}
+	}
+	void agregarAtencion(Atencion* atencion)
+	{
+		if (cantActual >= cantMax)
+		{
+			cout << "ya no entran mas atenciones" << endl;
+			return;
+		}
+		atenciones[cantActual] = atencion;
+		cantActual++;
+	}
+};
+class VectorPacientes
+{
+private:
+    Paciente** pacientes;  
+    int cantMax;         
+    int cantActual;          
+
+public:
+    VectorPacientes(int cantMax1)
+    {
+        cantMax = cantMax1;
+        cantActual = 0;
+        pacientes = new Paciente* [cantMax];
+    }
+    void agregarPaciente(Paciente* pa)
+    {
+        if (cantActual >= cantMax)
+        {
+            cout << "ya no entran mas nene" << endl;
+            return;
+        }
+        if (pa->getTipo() == "Externo")
+            pacientes[cantActual] = new PacienteExterno(pa->getNombre(), pa->getEdad());
+        else if (pa->getTipo() == "Asegurado")
+            pacientes[cantActual] = new PacienteAsegurado(pa->getNombre(), pa->getEdad());
```

Comment:
No es necesario crear un nuevo objeto
```suggestion
            pacientes[cantActual] = pa;
        else if (pa->getTipo() == "Asegurado")
            pacientes[cantActual] = pa;
```